### PR TITLE
feat(builder): add aggregator filter

### PR DESCRIPTION
**Pull request by yeshev**

### CODE REVIEW

File: frontend/src/container/QueryBuilder/filters/AggregatorFilter/AggregatorFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+			placeholder={`${transformToUpperCase(
+				query.dataSource,
+			)} Name (Start typing to get suggestions)`}
+			style={{ flex: 1, minWidth: 200 }}
+			showArrow={false}
+			filterOption={false}
+			onSearch={handleSearchAttribute}
+			notFoundContent={isFetching ? <Spin size="small" /> : null}
+			options={optionsData}
+			value={transformStringWithPrefix({
```
Comment (palashgdev) [inline frontend/src/container/QueryBuilder/filters/AggregatorFilter/AggregatorFilter.tsx:70]: wrap this one memo

File: frontend/src/container/QueryBuilder/filters/OperatorsSelect/OperatorsSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+		}),
+	);
+
+	return (
+		<Select
+			options={operatorsOptions}
+			value={value}
+			onChange={onChange}
+			// eslint-disable-next-line react/jsx-props-no-spreading
+			{...props}
```
Comment (palashgdev) [inline frontend/src/container/QueryBuilder/filters/OperatorsSelect/OperatorsSelect.tsx:29]: would it be possible not to spread

File: frontend/src/providers/QueryBuilder.tsx [hunk 3]
```diff
@@ -44,21 +51,51 @@ export function QueryBuilderProvider({
 	// ** TODO: type the params which will be used for request of the data for query builder
 
 	const [queryBuilderData, setQueryBuilderData] = useState<QueryBuilderData>({
-		queryData: [],
+		// ** TODO temporary initial value for first query for testing first filters
+		queryData: [
+			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+			// @ts-ignore
```
Comment (palashgdev) [inline frontend/src/providers/QueryBuilder.tsx:57]: i am hoping we would be removing this one

File: frontend/src/providers/QueryBuilder.tsx [hunk 4]
```diff
@@ -9,27 +10,33 @@ import React, {
...
+import { DataSource, MetricAggregateOperator } from 'types/common/queryBuilder';
 
 export type QueryBuilderData = {
-	queryData: IBuilderQuery[];
+	queryData: IBuilderQueryForm[];
 	queryFormulas: IBuilderFormula[];
 };
 
 // ** TODO: temporary types for context, fix it during development
 export type QueryBuilderContextType = {
```
Comment (palashgdev) [inline frontend/src/providers/QueryBuilder.tsx:23]: let's move this one in types

File: frontend/src/providers/QueryBuilder.tsx [hunk 5]
```diff
@@ -9,27 +10,33 @@ import React, {
...
 	handleSetFormulaData: (index: number, formulaData: IBuilderFormula) => void;
+	initQueryBuilderData: (queryBuilderData: QueryBuilderData) => void;
 };
 
 export const QueryBuilderContext = createContext<QueryBuilderContextType>({
 	queryBuilderData: { queryData: [], queryFormulas: [] },
 	resetQueryBuilderData: () => {},
 	handleSetQueryData: () => {},
 	handleSetFormulaData: () => {},
+	initQueryBuilderData: () => {},
```
Comment (palashgdev) [inline frontend/src/providers/QueryBuilder.tsx:39]:
```suggestion
	initQueryBuilderData: VoidFunction,
```

**Review by palashgdev — APPROVED** [review]

File: frontend/src/container/QueryBuilder/filters/OperatorsSelect/OperatorsSelect.tsx (hunk 2 above)
Comment (yeshev) [reply frontend/src/container/QueryBuilder/filters/OperatorsSelect/OperatorsSelect.tsx:29]: I created a reusable selector for the operators and we can pass as props any prop for native select + our props. Is the spread bad practice?

Comment (palashgdev) [reply frontend/src/container/QueryBuilder/filters/OperatorsSelect/OperatorsSelect.tsx:29]: we should be aware what props child is taking

**Review by palashgdev — APPROVED** [review]

**Pull request merged**